### PR TITLE
feat(web-app): add typesetstyle component

### LINKED ISSUE #662: TypesetStyle MDX component
https://carbondesignsystem.com/guidelines/typography/type-sets
```
<TypesetStyle typesets="smallStyle" />
<TypesetStyle typesets="body" />
<TypesetStyle typesets="fixedHeadings" />
<TypesetStyle typesets="fluidHeadings" breakpointControls={true} />
<TypesetStyle typesets="fluidCallouts,fluidDisplay" breakpointControls={true} />

### PR DESCRIPTION
**Pull request by alisonjoseph**
Closes #662

Add TypeSetStyle component, complete refactor of the code from the current website, tokenized all the colors, converted class->functional component. 

- known bug, type classes aren't working with correct prefix. Should be fixed in next release from Carbon https://github.com/carbon-design-system/carbon/pull/11228 Shouldn't block this PR, but we should update to latest Carbon version next sprint.

#### Testing / reviewing

http://localhost:3000/mdx/typesetstyle
http://localhost:6006/?path=/docs/platform-components-typesetstyle--body

### CODE REVIEW

File: services/web-app/components/typeset-style/typeset-example.js [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+        fontWeight: displayWeight(
+          currentBreakpointSpecs['font-weight'],
+          currentBreakpointSpecs['font-style']
+        ),
+        fontSize: `${`${calculateFluidTypeSize('font-size')}px / `}${currentBreakpointSpecs[
+          'font-size'
+        ]
+          .toString()
+          .replace('0.', '.')}rem`,
+        // eslint-disable-next-line no-useless-concat
```
Comment (francinelucca) [inline services/web-app/components/typeset-style/typeset-example.js:95]: is this necessary with our eslint rules?

File: services/web-app/components/typeset-style/typeset-example.js [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+        values(breakpoints).findIndex((width) => props.simulatedScreenWidth <= width)
+      )
+
+      const currentBreakpointPx = values(breakpoints)[indexOfClosestLargerBreakpoint]
+
+      const currentBreakpointName = findKey(breakpoints, (val) => val === currentBreakpointPx)
+      const getCurrentCompoundStylesForBreakpoint = (breakpointName) => {
+        const typeKeys = Object.keys(breakpoints)
+        const typeStylesUntilCurrentBreakpoint = []
+        // eslint-disable-next-line no-restricted-syntax
```
Comment (francinelucca) [inline services/web-app/components/typeset-style/typeset-example.js:41]: is this necessary with our eslint rules?

**Review by francinelucca — DISMISSED** [review]
LGTM, minor comments

Note: sticky control bar looks iffy on storybook:
image

File: services/web-app/components/typeset-style/typeset-example.js (hunk 1 above)
Comment (alisonjoseph) [reply services/web-app/components/typeset-style/typeset-example.js]: nope! good catch

**Comment by alisonjoseph** [comment]
@francinelucca I think that's ok, as long as they look correct on the mdx page.

**Review by francinelucca — DISMISSED** [review]

**Review by francinelucca — DISMISSED** [review]

**Review by andreancardona — APPROVED** [review]
all looks good tested in storybook!

**Pull request merged**